at-quannguyen/W4/main.rb

```diff
@@ -0,0 +1,621 @@
+    if number == false
+      puts 'input not string !'
+      csleep
+      cexit
+      action_login_register
+    end
+    action_register if number == 1
+    action_login if number == 2
+    action_forgot_password if number == 3
+  end
+
+  def action_login
+    clear
+    puts '========= Login ========='
+    print 'username : '
+    username = gets.chomp
+    puts ''
+    print 'password : '
+    password = gets.chomp
+
+    employee = Employee.new.tap do |e|
+      e.username = username
+      e.password = password
+    end
+
+    if @session_controller.login(employee)
+      puts 'Login complete !'
+      csleep
+      action_after_login
+    else
+      puts 'Login fail !'
+      csleep
+      cexit
+      action_login
+    end
+  end
+
+  def action_register
+    clear
+    puts '========= Register ========='
+    print 'name : '
+    name = gets.chomp
+    puts ''
+    print 'username : '
+    username = gets.chomp
+    puts ''
+    print 'password : '
+    password = gets.chomp
+
+    employee = Employee.new.tap do |e|
+      e.name = name
+      e.username = username
+      e.password = password
+      e.role = 0
+      e.leave_balance = 12
+    end
+
+    if @session_controller.register(employee)
+      puts 'Register complete !'
+      csleep
+      action_login
+    else
+      puts 'Username extis !'
+      csleep
+      cexit
+      action_register
+    end
+  end
+
+  def action_forgot_password
+    clear
+    puts '========= Forgot Password ========='
+    print 'username : '
+    username = gets.chomp
+    if @session_controller.username_by_employee(username).count.zero?
+      puts 'Account does not exist !'
+      csleep
+      cexit
+      action_forgot_password
+    else
+      clear
+      print 'new password : '
+      password = gets.chomp
+      employee = Employee.new.tap do |e|
+        e.username = username
+        e.password = password
+      end
+      if @session_controller.forgot_password(employee)
+        puts 'Action forgot complete !'
+        csleep
+        action_login
+      else
+        puts 'Action forgot extis !'
+        csleep
+        cexit
+        action_forgot_password
+      end
+    end
+  end
+
+  def action_after_login
+    clear
+    puts '========= Task Manager ========='
+    puts '1. Team management'
+    puts '2. Leave management'
+    puts '3. Notification management'
+    puts '4. CSV'
+    print 'Action : '
+    number = begin
+               Integer(gets.chomp)
+             rescue StandardError
+               false
+             end
+
+    if number == false
+      puts 'input not string !'
+      csleep
+      cexit
+      action_after_login
+    end
+
+    action_team_manager if number == 1
+    action_leave_manager if number == 2
+    action_notification_manager if number == 3
+    action_csv_manager if number == 4
+  end
+
+  def action_team_manager
+    clear
+    account = @session_controller.current_user
+    query = account.to_a[0]
+    team_id = query['team_id']
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/7deb92b5e1c7275222ca44a289eebcc27cbc38fe/at-quannguyen/W4/main.rb#L168"></span>Lint/UselessAssignment: Useless assignment to variable - <code>team_id</code>. (https://github.com/rubocop-hq/ruby-style-guide#underscore-unused-vars)</td>
    </tr>
  </tbody>
</table>